walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java

```diff
@@ -14,10 +14,121 @@
 
 package com.google.sps;
 
+import java.util.Arrays;
+import java.util.ArrayList;
 import java.util.Collection;
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Set;
+
+import java.lang.*;
 
 public final class FindMeetingQuery {
+  /** MeetingRequest request has:
+      getAttendees() method which returns hashSet of mandatory attendees
+      getDuration() method which tells us how long our meeting has to be */
+
+  /** Collection<Event> events has events. Each event has:
+      getWhen() method which tells us time range of event
+      getAttendees() method which tells us who has to go to each event */
+
   public Collection<TimeRange> query(Collection<Event> events, MeetingRequest request) {
```

Comment:
The function is pretty big by general standards, and there is scope to make it smaller by adding 2-3 helper functions.